Sources/NIOCore/EventLoop.swift

```diff
@@ -777,13 +753,14 @@ extension EventLoopGroup {
     }
 
     public func syncShutdownGracefully() throws {
-        if let eventLoop = MultiThreadedEventLoopGroup.currentEventLoop {
-            preconditionFailure("""
-            BUG DETECTED: syncShutdownGracefully() must not be called when on an EventLoop.
-            Calling syncShutdownGracefully() on any EventLoop can lead to deadlocks.
-            Current eventLoop: \(eventLoop)
-            """)
-        }
+        // FIXME: Add new API surface to reimplement this.
```

Comment:
Are the fixes imminent or do we need to create issues for them?

Reply:
The fixes should be part of this patch series, yeah, as otherwise we're removing functionality in many of these cases.

Reply:
#1899 #1900

Reply:
I guess we can remove this FIXME's now.

Reply:
Nope, this _particular_ FIXME is actually still open. On me to address.